src/lib/route-broadcaster.js

```diff
@@ -0,0 +1,126 @@
+   * @param {Backend} backend
+   * @param {Object} config
+   * @param {Object} config.ledgerCredentials
+   * @param {Object} config.tradingPairs
+   * @param {Number} config.minMessageWindow
+   */
+  constructor (routingTables, backend, config) {
+    this.baseURI = routingTables.baseURI
+    this.routingTables = routingTables
+    this.backend = backend
+    this.ledgerCredentials = config.ledgerCredentials
+    this.tradingPairs = config.tradingPairs
+    this.minMessageWindow = config.minMessageWindow
+    this.adjacentConnectors = {}
+    this.adjacentLedgers = {}
+    for (const pair of config.tradingPairs) {
+      const destinationLedger = pair[1].split('@')[1]
+      this.adjacentLedgers[destinationLedger] = true
+    }
+  }
+
+  * start () {
+    yield this.crawlLedgers()
+    yield this.reloadLocalRoutes()
+    yield this.broadcast()
+    setInterval(() => this.routingTables.removeExpiredRoutes(), 1000)
+    defer.setInterval(() => {
+      return this.reloadLocalRoutes().then(this.broadcast.bind(this))
+    }, BROADCAST_INTERVAL)
+  }
+
+  broadcast () {
+    const routes = this.routingTables.toJSON()
+    return Promise.all(
+      Object.keys(this.adjacentConnectors).map(
+        (adjacentConnector) => this._broadcastTo(adjacentConnector, routes)))
+  }
+
+  _broadcastTo (adjacentConnector, routes) {
+    return request({
+      method: 'POST',
+      uri: adjacentConnector + '/routes',
+      body: routes,
+      json: true
+    }).then((res) => {
+      if (res.statusCode !== 200) {
+        throw new Error('Unexpected status code: ' + res.statusCode)
+      }
+    })
+  }
+
+  crawlLedgers () {
+    return Object.keys(this.adjacentLedgers).map(this._crawlLedger, this)
+  }
+
+  * _crawlLedger (ledger) {
+    const res = yield request({
+      method: 'GET',
+      uri: ledger + '/connectors',
+      json: true
+    })
+    if (res.statusCode !== 200) {
+      throw new Error('Unexpected status code: ' + res.statusCode)
+    }
+    const connectors = _.map(res.body, 'connector')
+    for (const connector of connectors) {
+      // Don't broadcast routes to ourselves.
+      if (connector === this.baseURI) continue
+      this.adjacentConnectors[connector] = true
+    }
+  }
+
+  /**
+   * @returns {Promise}
+   */
+  reloadLocalRoutes () {
+    return this._getLocalRoutes().then(
+      (routes) => this.routingTables.addLocalRoutes(routes))
+  }
+
+  _getLocalRoutes () {
+    return Promise.all(this.tradingPairs.map((pair) => {
+      return this._tradingPairToQuote(pair)
+        .then((quote) => this._quoteToLocalRoute(quote))
+    }))
+  }
+
+  _tradingPairToQuote (pair) {
+    const sourceLedger = pair[0].split('@')[1]
+    const destinationLedger = pair[1].split('@')[1]
+    // TODO change the backend API to return curves, not points
+    return co(this.backend.getQuote.bind(this.backend), {
+      source_ledger: sourceLedger,
+      destination_ledger: destinationLedger,
+      source_amount: 100000000
```

Comment:
It creates a curve `[[0, 0], [100000000, fx(100000000)]]`. It is just a workaround until the backends are rewritten to return the curve directly.